api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -507,4 +508,25 @@ public String toString() {
                 .map(this::identifierFieldToString)
                 .collect(Collectors.toList())));
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
More on the style, I think this is easier to read:
```suggestion
    if (o instanceof Schema) {
```

Reply:
We need to call `getClass() != o.getClass()` because `o` maybe subclass of `Schema`